java/compaction/compaction-core/src/test/java/sleeper/compaction/testutils/InMemoryCompactionJobStatusStoreTest.java

```diff
@@ -351,9 +373,11 @@ void shouldGetJobInTimePeriodWhenFirstRunIsInPeriod() {
                             jobStatusFrom(records().fromUpdates(
```

Comment:
The fact that both runs are uncommitted invalidates this test. It specifies that only one run is in the period, but they actually both are, because when a job is uncommitted it's in any period that takes place after it. Both of the runs need to be committed to keep this test behaviourally the same.